apps/api/src/app/tenant/tenant.controller.ts

```diff
@@ -0,0 +1,178 @@
+import { GetTenantCommand } from './usecases/get-tenant/get-tenant.command';
+import { UpdateTenantResponseDto } from './dtos/update-tenant-response.dto';
+import { UpdateTenantRequestDto } from './dtos/update-tenant-request.dto';
+import { UpdateTenant } from './usecases/update-tenant/update-tenant.usecase';
+import { DeleteTenantCommand } from './usecases/delete-tenant/delete-tenant.command';
+import { DeleteTenant } from './usecases/delete-tenant/delete-tenant.usecase';
+import { UpdateTenantCommand } from './usecases/update-tenant/update-tenant.command';
+import { ApiOkPaginatedResponse } from '../shared/framework/paginated-ok-response.decorator';
+import { PaginatedResponseDto } from '../shared/dtos/pagination-response';
+import { GetTenantsRequestDto } from './dtos/get-tenants.request.dto';
+import { GetTenants } from './usecases/get-tenants/get-tenants.usecase';
+import { GetTenantsCommand } from './usecases/get-tenants/get-tenants.command';
+
+@Controller('/tenants')
+@ApiTags('Tenants')
+@UseInterceptors(ClassSerializerInterceptor)
+@UseGuards(JwtAuthGuard)
+@ApiExcludeController()
+export class TenantController {
+  constructor(
+    private createTenantUsecase: CreateTenant,
+    private updateTenantUsecase: UpdateTenant,
+    private getTenantUsecase: GetTenant,
+    private deleteTenantUsecase: DeleteTenant,
+    private getTenantsUsecase: GetTenants
+  ) {}
+
+  @Get('')
+  @ExternalApiAccessible()
+  @UseGuards(JwtAuthGuard)
+  @ApiOkPaginatedResponse(GetTenantResponseDto)
+  @ApiOperation({
+    summary: 'Get tenants',
+    description: 'Returns a list of tenants, could paginated using the `page` and `limit` query parameter',
+  })
+  async getTenants(
+    @UserSession() user: IJwtPayload,
+    @Query() query: GetTenantsRequestDto
+  ): Promise<PaginatedResponseDto<GetTenantResponseDto>> {
+    return await this.getTenantsUsecase.execute(
+      GetTenantsCommand.create({
+        organizationId: user.organizationId,
+        environmentId: user.environmentId,
+        page: query.page ? Number(query.page) : 0,
+        limit: query.limit ? Number(query.limit) : 10,
+      })
+    );
+  }
+
+  @Get('/:identifier')
+  @ApiResponse(GetTenantResponseDto)
+  @ApiOperation({
+    summary: 'Get tenant',
+    description: `Get tenant by your internal id used to identify the tenant`,
+  })
+  @ExternalApiAccessible()
+  getTenantById(
+    @UserSession() user: IJwtPayload,
+    @Param('identifier') identifier: string
+  ): Promise<GetTenantResponseDto> {
+    return this.getTenantUsecase.execute(
+      GetTenantCommand.create({
+        environmentId: user.environmentId,
+        organizationId: user.organizationId,
+        identifier: identifier,
+      })
+    );
+  }
+
+  @Post('/')
+  @ExternalApiAccessible()
+  @ApiResponse(CreateTenantResponseDto)
+  @ApiOperation({
+    summary: 'Create tenant',
+    description: 'Create tenant under the current environment',
+  })
+  @ApiConflictResponse({
```

Comment:
👏🏻 
(I don't recall seeing a test for this though)

Reply:
true, fixed existing one while i wend over your comments 🔥

Reply:
true, fixed an existing one while i went over your comments 🔥